### PR TITLE
IW-993 | support choosing suggestion

### PR DESCRIPTION
**Pull request by latata**
@xkxd @hakubo

### CODE REVIEW

File: src/components/HyperlinkingWrapper.js [hunk 1]
```diff
@@ -102,14 +103,15 @@ export default class HyperlinkingWrapper extends Component {
                 const [blotToEdit, blotRange] = this.getBlotFromIndex(range.index);
 
                 if (blotToEdit.statics.blotName === 'link') {
-                    const linkValue = blotToEdit.formats().link;
+                    const { url: linkValue, title: linkTitle } = blotToEdit.formats().link;
```
Comment (hakubo) [inline src/components/HyperlinkingWrapper.js]: Maybe linkHref?

File: src/components/InputTooltip.js [hunk 2]
```diff
@@ -104,34 +135,45 @@ class InputTooltip extends Component {
...
-        const { onAccept } = this.props;
-        const trimmedUrl = url.trim();
-
-        if (this.isValidUrl(trimmedUrl)) {
-            onAccept(trimmedUrl);
-        } else {
-            this.setState({ isLinkInvalid: true });
-        }
+    setSuggestionsListClicked(value) {
+        this.suggestionsListClicked = value;
```
Comment (hakubo) [inline src/components/InputTooltip.js]: Why this does not go to state?

File: src/components/InputTooltip.js [hunk 3]
```diff
@@ -104,34 +135,45 @@ class InputTooltip extends Component {
...
     }
 
     splitText(textToSplit, query) {
-        const escapedQuery = query.trim().toLowerCase().replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&');
-        const matchedIndex = textToSplit.toLowerCase().indexOf(escapedQuery);
+        const escapedQuery = query.trim()
+            .toLowerCase()
+            .replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&');
```
Comment (hakubo) [inline src/components/InputTooltip.js]: :beer: if you can tell me what it matches in months 😏🍻 can you add a comment please?

File: src/components/InputTooltip.js (hunk 2 above)
Comment (latata) [reply src/components/InputTooltip.js]: setState updates state asynchronously, so I think it's safer to set it like that.

Comment (xkxd) [reply src/components/InputTooltip.js]: is there any situation where having this in state may be problematic? what about using setState's callback?

**Review by hakubo — APPROVED** [review]

File: src/components/InputTooltip.js [hunk 4]
```diff
@@ -182,15 +251,19 @@ class InputTooltip extends Component {
...
+                    {isLinkInvalid && <span className="wds-input__hint">{i18n['hyperlinking-error']}</span>}
                 </div>
-                {linkValue && (
-                    <div className="pe-input-tooltip__suggestions wds-dropdown__content wds-is-not-scrollable">
+                {linkHref && (
+                    <div
+                        className="pe-input-tooltip__suggestions wds-dropdown__content wds-is-not-scrollable"
+                        role="presentation"
+                        onMouseDown={this.setSuggestionsListClicked.bind(this, true)}
+                        onMouseUp={this.setSuggestionsListClicked.bind(this, false)}
```
Comment (xkxd) [inline src/components/InputTooltip.js:265]: binding function in `render` is basically equal to `() => this.setSuggestionsListClicked()` so can we use arrow here?

**Review by xkxd — APPROVED** [review]

**Pull request merged**